functions/go/set-labels/transformer/labels.go

```diff
@@ -99,78 +75,365 @@ func (p *LabelTransformer) Transform(objects fn.KubeObjects) error {
 		return nil
 	}
 	for _, o := range objects.WhereNot(func(o *fn.KubeObject) bool { return o.IsLocalConfig() }) {
-		// this label need to set for all GVK
-		metaLabelsPath := FieldPath{"metadata", "labels"}
-		updatedLabels, err := updateLabels(&o.SubObject, metaLabelsPath, p.NewLabels, true)
+		err := func() error {
+			// set meta labels
+			if err := p.setObjectMeta(o); err != nil {
+				return err
+			}
+			// set selector
+			if err := p.setSelector(o); err != nil {
+				return err
+			}
+			// set PodTemplateSpec
+			if err := p.setPodTemplateSpec(o); err != nil {
+				return err
+			}
+			// set other corner cases
+			if err := p.setVolumeClaimTemplates(o); err != nil {
+				return err
+			}
+			if err := p.setJobTemplateSpecMeta(o); err != nil {
+				return err
+			}
+			if err := p.setNetworkPolicyRule(o); err != nil {
+				return err
+			}
+			return nil
+		}()
 		if err != nil {
 			return err
 		}
-		p.LogResult(o, metaLabelsPath, updatedLabels)
-		// set other common labels according to specific GVK
-		err = p.setLabelsInSpecs(o)
-		if err != nil {
+		p.LogResult(o, p.ResultCount)
+	}
+	return nil
+}
+
+// hasJobTemplateSpec check if the object is CronJob, this kind might have JobTemplateSpec, which contains label fieldPath
+func hasJobTemplateSpec(o *fn.KubeObject) bool {
+	return o.IsGVK("batch", "", "CronJob")
+}
+
+// setJobTemplateSpecMeta set MetaObject in struct JobTemplateSpec
+func (p *LabelTransformer) setJobTemplateSpecMeta(o *fn.KubeObject) error {
+	if hasJobTemplateSpec(o) {
```

Comment:
I think having this check inside the `setJobTemplateSpecMeta` is a little bit confusing, the name of the method `setJobTemplateSpecMeta` seems to imply that we already know that the object is a Job. 

I think a better structure would be to change lines 90-99 above to something like

```
if hasJobTempalteSpec(o) {
  p.setJobTemplateSpecMeta
}
```

and same for `p.setVolumeClaimTemplates(o)` and `p.setNetworkPolicyRule(o)`

Reply:
I see. I left this change because for functions like `setSelector`, I need to check if it has different selector type inside the function. So for consistency I just left all `hasXXX` function inside `setXXX`. But for the corner cases, it seems a good idea to separate them out.